lib/ansible/modules/files/find.py

```diff
@@ -440,12 +440,30 @@ def main():
                                 r['checksum'] = module.sha1(fsname)
                             filelist.append(r)
 
+                    elif stat.S_ISBLK(st.st_mode) and params['file_type'] == 'block':
+                        if pfilter(fsobj, params['patterns'], params['excludes'], params['use_regex']) and agefilter(st, now, age, params['age_stamp']):
+
+                            r.update(statinfo(st))
+                            filelist.append(r)
+
+                    elif stat.S_ISCHR(st.st_mode) and params['file_type'] == 'char':
+                        if pfilter(fsobj, params['patterns'], params['excludes'], params['use_regex']) and agefilter(st, now, age, params['age_stamp']):
+
+                            r.update(statinfo(st))
+                            filelist.append(r)
```

Comment:
since all new options are the same as 'directory' when it comes to the subsequent filter matching, just change it into one elif entry with `file_type in  ('directory', 'char' ...)`

Reply:
That was my first setup until I saw link was also separate. Do you think we can drop the stat.IS* check and move this to be the last elif?

Reply:
link used to be different, not sure when it got made the same

Reply:
@bcoca did you had a chance to review the modification. To be honest, their isn't really a very elegant way to fix it imho.